### PR TITLE
EASY-2465: fix 7 vulnerabilities and dependency upgrades

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes EASY-2465

#### When applied it will
* fix 7 vulnerabilities, amongst others the XSS vulnerability in `serialize-javascript`
* dependency upgrades

#### The way to check/fix vulnerabilities
* `npm audit` - lists current vulnerabilities
* `npm audit fix` - fix these vulnerabilities

@DANS-KNAW/easy for review

### CODE REVIEW

**Comment by janvanmansum** [comment]
Are all the vulnerabilities in transitive dependencies of `express-fileupload`?

**Comment by rvanheest** [comment]
> Are all the vulnerabilities in transitive dependencies of `express-fileupload`?

Don't think so, @janvanmansum. It's more than that. For one, `serialize-javascript` is in there as well...

**Comment by rvanheest** [comment]
@janvanmansum this is the listing from `npm audit` on the first commit:

```
npm audit

                       === npm audit security report ===

# Run  npm install react-scripts@3.3.0  to resolve 1 vulnerability
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ serialize-javascript                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > terser-webpack-plugin > serialize-javascript │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1426                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


# Run  npm update handlebars --depth 7  to resolve 5 vulnerabilities
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ handlebars                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > jest > jest-cli > @jest/core >               │
│               │ @jest/reporters > istanbul-reports > handlebars              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1164                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Denial of Service                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ handlebars                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > jest > jest-cli > @jest/core >               │
│               │ @jest/reporters > istanbul-reports > handlebars              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1300                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary Code Execution                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ handlebars                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > jest > jest-cli > @jest/core >               │
│               │ @jest/reporters > istanbul-reports > handlebars              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1316                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary Code Execution                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ handlebars                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > jest > jest-cli > @jest/core >               │
│               │ @jest/reporters > istanbul-reports > handlebars              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1324                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ handlebars                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > jest > jest-cli > @jest/core >               │
│               │ @jest/reporters > istanbul-reports > handlebars              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1325                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


# Run  npm update terser-webpack-plugin --depth 3  to resolve 1 vulnerability
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Cross-Site Scripting                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ serialize-javascript                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > webpack > terser-webpack-plugin >            │
│               │ serialize-javascript                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1426                            │
└───────────────┴──────────────────────────────────────────────────────────────┘


found 7 vulnerabilities (3 moderate, 4 high) in 903773 scanned packages
  run `npm audit fix` to fix 7 of them.
```

File: webpack/base.config.js [hunk 1]
```diff
@@ -92,6 +95,7 @@ module.exports = (env, argv) => ({
                         loader: 'file-loader',
                         options: {
                             name: 'app/helptexts/[name].[ext]',
+                            esModule: false,
```
Comment (janvanmansum) [inline webpack/base.config.js]: What does this `esModule` field do?

Comment (rvanheest) [reply webpack/base.config.js]: That's a new field in `file-loader` version `5.x`. It's by default set to `true`, but for some reason it has to be `false` in our settings. See https://github.com/webpack-contrib/file-loader#esmodule and https://github.com/webpack-contrib/file-loader/issues/350#issuecomment-558260039.

**Pull request merged**